lib/es6/toolset.js

```diff
@@ -60,8 +60,14 @@ Toolset.prototype.initializePosix = function (install) {
         this.cCompilerPath = "gcc";
     }
 
+    if( this.options.generator ){
+        if (install) {
+            this.log.info("TOOL", "Using " + this.options.generator + " generator, as specified from commandline.");
+        }
+        this.generator = this.options.generator;
```

Comment:
Why do you set this.generator again? It's already have options.generator's value at this point.